plugins/performance-lab/includes/site-health/audit-enqueued-assets/hooks.php

```diff
@@ -13,107 +13,104 @@
+			'headers' => array(
+				'Accept'        => 'text/html',
+				'Cache-Control' => 'no-cache',
+			),
+		)
+	);
+
+	if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+		return;
+	}
 
-		foreach ( $wp_scripts->done as $handle ) {
-			$script = $wp_scripts->registered[ $handle ];
+	$html = wp_remote_retrieve_body( $response );
+	if ( '' === $html ) {
+		return;
+	}
+
+	$assets = array(
+		'scripts' => array(),
+		'styles'  => array(),
+	);
+
+	$processor = new WP_HTML_Tag_Processor( $html );
 
-			if ( ! $script->src || false !== strpos( $script->src, 'wp-includes' ) ) {
+	while ( $processor->next_tag() ) {
+		$tag = $processor->get_tag();
+
+		if ( 'SCRIPT' === $tag ) {
+			$src = $processor->get_attribute( 'src' );
+			if ( ! is_string( $src ) ) {
 				continue;
 			}
 
-			// Add any extra data (inlined) that was passed with the script.
-			$inline_size = 0;
-			if (
-				isset( $script->extra['after'] ) &&
-				is_array( $script->extra['after'] )
-			) {
-				foreach ( $script->extra['after'] as $extra ) {
-					$inline_size += ( is_string( $extra ) ) ? mb_strlen( $extra, '8bit' ) : 0;
-				}
-			}
+			// Note that when the "type" attribute is absent or empty, the element is treated as a classic JavaScript script.
+			$type = $processor->get_attribute( 'type' );
 
-			$path = perflab_aea_get_path_from_resource_url( $script->src );
-			if ( '' === $path ) {
+			// Skip external script with "async" or "defer" attributes.
+			if ( null !== $processor->get_attribute( 'async' ) || null !== $processor->get_attribute( 'defer' ) ) {
 				continue;
 			}
 
-			$enqueued_scripts[] = array(
-				'src'  => $script->src,
-				'size' => wp_filesize( $path ) + $inline_size,
-			);
-
-		}
-		set_transient( 'aea_enqueued_front_page_scripts', $enqueued_scripts, 12 * HOUR_IN_SECONDS );
-	}
-}
-add_action( 'wp_footer', 'perflab_aea_audit_enqueued_scripts', PHP_INT_MAX );
+			// Skip external script with a "type" attribute set to "module" as they are deferred by default.
+			if ( is_string( $type ) && '' !== $type && 'module' === strtolower( $type ) ) {
+				continue;
+			}
 
-/**
- * Audit enqueued and printed styles in the frontend. Ignore /wp-includes styles.
- *
- * It will save information in a transient for 12 hours.
- *
- * @since 1.0.0
- *
- * @global WP_Styles $wp_styles The WP_Styles current instance.
- */
-function perflab_aea_audit_enqueued_styles(): void {
-	if ( ! is_admin() && is_front_page() && current_user_can( 'view_site_health_checks' ) && false === get_transient( 'aea_enqueued_front_page_styles' ) ) {
-		global $wp_styles;
-		$enqueued_styles = array();
-		foreach ( $wp_styles->done as $handle ) {
-			$style = $wp_styles->registered[ $handle ];
-
-			if ( ! $style->src || false !== strpos( $style->src, 'wp-includes' ) ) {
+			// Skip external script with a "type" attribute that is not JavaScript.
+			if ( is_string( $type ) && '' !== $type && 'text/javascript' !== strtolower( $type ) ) {
```

Comment:
```suggestion
			if ( 'text/javascript' !== strtolower( (string) $type ) ) {
```

Reply:
Here, it is necessary to keep `'' !== $type` since an empty string value for `type` is considered valid in JavaScript.

Then it can be done like this:
```suggestion
			if ( '' !== $type && 'text/javascript' !== strtolower( (string) $type ) ) {
```

Reply:
Okay, this change breaks things because the `get_attribute` method returns `null` when the attribute is not present. Therefore, a `null !== $type` check or the previous `is_string( $type )` check needs to be added explicitly.

which check would be better?

Reply:
> Here, it is necessary to keep `'' !== $type` since an empty string value for `type` is considered valid in JavaScript.

Right, this is executed:

```html
<script type>document.write('hello');</script>
```

There's also this code in core which is relevant here:

https://github.com/WordPress/wordpress-develop/blob/27233a0a3b00cc5d683080722711bca15970fe8c/src/wp-includes/script-loader.php#L2948-L2958

With the HTML API, the tag would be JavaScript if:

1. The `type` attribute is absent (`$type === null`)
2. The `type` attribute is boolean and has no value (`$type === true`)
3. The `type` attribute is an empty string (`$type === ''`), but not if the value has whitespace in which case it is _not_ executed.
4. The `type` attribute contains `javascript`, `ecmascript`, `jscript`, or `livescript`.

Reply:
Done in https://github.com/WordPress/performance/pull/2059/commits/fc7a9a29995d9eb794a665683c28035d0b7e33f6.